src/Core/Addon/AddonManagerHelper.php

```diff
@@ -0,0 +1,306 @@
+			return false;
+		}
+
+		$this->logger->debug("Addon {addon}: {action}", ['action' => 'install', 'addon' => $addonId]);
+
+		$timestamp = @filemtime($addon_file_path);
+
+		@include_once($addon_file_path);
+
+		if (function_exists($addonId . '_install')) {
+			$func = $addonId . '_install';
+			$func();
+		}
+
+		$this->config->set('addons', $addonId, [
+			'last_update' => $timestamp,
+			'admin'       => function_exists($addonId . '_addon_admin'),
+		]);
+
+		if (!$this->isAddonEnabled($addonId)) {
+			$this->addons[] = $addonId;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Uninstalls an addon.
+	 *
+	 * @param string $addonId name of the addon
+	 */
+	public function uninstallAddon(string $addonId): void
+	{
+		$addonId = Strings::sanitizeFilePathItem($addonId);
+
+		$this->logger->debug("Addon {addon}: {action}", ['action' => 'uninstall', 'addon' => $addonId]);
+		$this->config->delete('addons', $addonId);
+
+		$addon_file_path = $this->getAddonPath() . '/' . $addonId . '/' . $addonId . '.php';
+
+		@include_once($addon_file_path);
+
+		if (function_exists($addonId . '_uninstall')) {
+			$func = $addonId . '_uninstall';
+			$func();
+		}
+
+		// Remove registered hooks for the addon
+		// Handles both relative and absolute file paths
+		$condition = ['`file` LIKE ?', "%/$addonId/$addonId.php"];
+
+		$result = $this->database->delete('hook', $condition);
+
+		if ($result) {
+			$this->cache->delete('routerDispatchData');
+		}
+
+		unset($this->addons[array_search($addonId, $this->addons)]);
+	}
+
+	/**
+	 * Load addons.
+	 *
+	 * @internal
+	 */
+	public function loadAddons(): void
+	{
+		$this->addons = array_keys(array_filter($this->config->get('addons') ?? []));
+	}
+
+	/**
+	 * Reload (uninstall and install) all installed and modified addons.
+	 */
+	public function reloadAddons(): void
+	{
+		$addons = array_filter($this->config->get('addons') ?? []);
+
+		foreach ($addons as $addonName => $data) {
+			$addonId = Strings::sanitizeFilePathItem(trim($addonName));
+
+			$addon_file_path = $this->getAddonPath() . '/' . $addonId . '/' . $addonId . '.php';
+
+			if (!file_exists($addon_file_path)) {
+				continue;
+			}
+
+			if (array_key_exists('last_update', $data) && intval($data['last_update']) === filemtime($addon_file_path)) {
+				// Addon unmodified, skipping
+				continue;
+			}
+
+			$this->logger->debug("Addon {addon}: {action}", ['action' => 'reload', 'addon' => $addonId]);
+
+			$this->uninstallAddon($addonId);
+			$this->installAddon($addonId);
+		}
+	}
+
+	/**
+	 * Get the comment block of an addon as value object.
+	 */
+	public function getAddonInfo(string $addonId): AddonInfo
+	{
+		$default = [
+			'id'   => $addonId,
+			'name' => $addonId,
+		];
+
+		if (!is_file($this->getAddonPath() . "/$addonId/$addonId.php")) {
+			return AddonInfo::fromArray($default);
+		}
+
+		$this->profiler->startRecording('file');
+
+		$raw = file_get_contents($this->getAddonPath() . "/$addonId/$addonId.php");
```

Comment:
Can `$raw` be null? If so plz check & unittest it :)

Reply:
hm .. but we would block the administration of the node in case there's a buggy addon, won't we? I'd prefer logging a warning, but don't throw an exception to the user

Reply:
Thats ok :)